openedx/core/djangoapps/content_libraries/tasks.py

```diff
@@ -169,6 +174,70 @@ def duplicate_children(
                 self.status.fail({'raw_error_msg': str(exception)})
 
 
+@shared_task(base=LoggedTask)
+@set_code_owner_attribute
+def create_or_update_xblock_upstream_link(usage_key):
+    """
+    Create or update upstream link for a single xblock.
+    """
+    ensure_cms("create_or_update_xblock_upstream_link may only be executed in a CMS context")
+    xblock = modulestore().get_item(UsageKey.from_string(usage_key))
+    if not xblock.upstream or not xblock.upstream_version:
+        TASK_LOGGER.info(f"No upstream or upstream_version found for xblock: {xblock.usage_key}")
+        return
+    try:
+        course_name = CourseOverview.get_from_id(xblock.course_id).display_name_with_default
+    except CourseOverview.DoesNotExist:
+        TASK_LOGGER.exception(f'Could not find course: {xblock.course_id}')
```

Comment:
Do we need to log the whole exception? I think an error is enough:
```suggestion
        TASK_LOGGER.error(f'Could not find course: {xblock.course_id}')
```

Reply:
This should be possible hence logging the whole exception.